integration-tests/docker/cluster.go

```diff
@@ -0,0 +1,117 @@
+package docker
+
+import (
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	ctfClient "github.com/smartcontractkit/chainlink-testing-framework/client"
+	"github.com/smartcontractkit/chainlink-testing-framework/logwatch"
+	"github.com/smartcontractkit/chainlink/integration-tests/client"
+	"github.com/smartcontractkit/chainlink/integration-tests/contracts"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+/*
+	These are the high level components you should reuse in your docker tests in other repos
+	Interfaces from env.go should be moved to chainlink-env or CTF in the next stage
+	Implementations must stay in the repos where they belong
+*/
+
+type Endpoints struct {
+	Networks   []string
+	Nodes      []string
+	Mockserver string
+}
+
+type Clients struct {
+	Networks         []blockchain.EVMClient
+	NetworkDeployers []contracts.ContractDeployer
+	Mockserver       *ctfClient.MockserverClient
+	Chainlink        []*client.Chainlink
+}
+
+// ConnectClients connects network/CL nodes/mockserver clients
+func ConnectClients(network blockchain.EVMNetwork, env *Environment) (*Clients, error) {
```

Comment:
@skudasov why not put each client in the container wrapper like docker/mockserver.go? This would be much simpler then. See my code in Mercury https://github.com/smartcontractkit/mercury-server/blob/main/e2e_tests/docker/test_env/mockserver.go#L18-L23

Reply:
It seems like a natural way to group those test env components

Reply:
What if you need another client? I think deployments should be decoupled from clients because sometimes I'm using different clients. What will we do if we require a Solana client? @lukaszcl

Reply:
I think that we should make deployments reusable *and* decoupled from clients.
In e2e tests, we usually only operate with `clients` that are connecting to `environment` through `URLs`
I'd like this decoupling much more.